src/Microsoft.Sbom.Api/PackageDetails/ComponentDetailsUtils/RubyGemsUtils.cs

```diff
@@ -0,0 +1,291 @@
+        this.log = log ?? throw new ArgumentNullException(nameof(log));
+        this.recorder = recorder ?? throw new ArgumentNullException(nameof(recorder));
+
+        this.rubyGemsPath = GetRubyGemsSpecificationsPath();
+    }
+
+    // Takes in a scanned component and attempts to find the associated gemspec file. If it is not found then it returns null.
+    public string? GetMetadataLocation(ScannedComponent scannedComponent)
+    {
+        var gemspecLocation = rubyGemsPath;
+
+        if (string.IsNullOrEmpty(gemspecLocation))
+        {
+            return null;
+        }
+
+        var componentName = scannedComponent.Component.PackageUrl?.Name.ToLower();
+        var componentVersion = scannedComponent.Component.PackageUrl?.Version;
+
+        var fullGemspecPath = Path.Join(gemspecLocation, $"{componentName}-{componentVersion}.gemspec");
+
+        if (fileSystemUtils.FileExists(fullGemspecPath))
+        {
+            return fullGemspecPath;
+        }
+        else
+        {
+            // Enumerate directories in the gemspec location and then search for the gemspec file in each directory
+            var directories = fileSystemUtils.GetDirectories(gemspecLocation);
+
+            foreach (var directory in directories)
+            {
+                fullGemspecPath = Path.Join(directory, $"{componentName}-{componentVersion}.gemspec");
+
+                if (fileSystemUtils.FileExists(fullGemspecPath))
+                {
+                    return Path.GetFullPath(fullGemspecPath);
+                }
+            }
+        }
+
+        return null;
+    }
+
+    public ParsedPackageInformation ParseMetadata(string gemspecLocation)
+    {
+        var supplierField = string.Empty;
+        var licenseField = string.Empty;
+
+        // Regex patterns for finding the relevant properties of the .gemspec file.
+        var namePattern = @"s.name\s*=\s*""([^""]+)""";
+        var versionPattern = @"s.version\s*=\s*""([^""]+)""";
+        var authorsPattern = @"s.authors\s*=\s*\[([^\]]+)\]";
+        var licensesPattern = @"s\.licens(?:e|es)\s*=\s*\[([^\]]+)\]";
+
+        try
+        {
+            if (!fileSystemUtils.FileExists(gemspecLocation))
+            {
+                throw new PackageMetadataParsingException("Gemspec file not found.");
+            }
+
+            var fileContent = fileSystemUtils.ReadAllText(gemspecLocation);
+
+            var name = GetMatchesFromPattern(fileContent, namePattern);
+            var version = GetMatchesFromPattern(fileContent, versionPattern);
+
+            // We only take the first author to maintain consistency with what we do with the other component types.
+            var supplierList = GetMatchesFromPattern(fileContent, authorsPattern);
+
+            if (supplierList.Any())
+            {
+                supplierField = supplierList.First().Replace("\".freeze", string.Empty);
```

Comment:
Ruby has a required authors field that looks like "s.authors = ["Nobu Nakada".freeze]" After reading the value I get something like "Nobu Nakada\".freeze" so even if the name was "anti.freeze it would be fine because we would end up with "anti.freeze\".freeze"

Reply:
Ah, I missed the `\"` in the front.